testing/integration/beats_serverless_test.go

```diff
@@ -0,0 +1,589 @@
+
+	// the require.* code will fail without these, so assume the values are non-nil
+	runner.ESHost = os.Getenv("ELASTICSEARCH_HOST")
+	runner.user = os.Getenv("ELASTICSEARCH_USERNAME")
+	runner.pass = os.Getenv("ELASTICSEARCH_PASSWORD")
+	runner.kibHost = os.Getenv("KIBANA_HOST")
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+	defer cancel()
+
+	beatOutConfig := `
+output.elasticsearch:
+  hosts: ["%s"]
+  api_key: "%s:%s"
+setup.kibana:
+  host: %s
+metricbeat.config.modules:
+  path: ${path.config}/modules.d/*.yml
+processors:
+  - add_fields:
+      target: host
+      fields:
+        test-id: %s
+`
+	if runner.testbeatName == "filebeat" {
+		beatOutConfig = `
+output.elasticsearch:
+  hosts: ["%s"]
+  api_key: "%s:%s"
+setup.kibana:
+  host: %s
+filebeat.config.modules:
+  - modules: system
+    syslog:
+      enabled: true
+    auth:
+      enabled: true
+processors:
+  - add_fields:
+      target: host
+      fields:
+        test-id: %s
+`
+	}
+
+	apiResp, err := estools.CreateAPIKey(ctx, runner.requirementsInfo.ESClient, estools.APIKeyRequest{Name: "test-api-key", Expiration: "1d"})
+	require.NoError(runner.T(), err)
+
+	// beats likes to add standard ports to URLs that don't have them, and ESS will sometimes return a URL without a port, assuming :443
+	// so try to fix that here
+	fixedKibanaHost := runner.kibHost
+	parsedKibana, err := url.Parse(runner.kibHost)
+	require.NoError(runner.T(), err)
+	if parsedKibana.Port() == "" {
+		fixedKibanaHost = fmt.Sprintf("%s:443", fixedKibanaHost)
+	}
+
+	fixedESHost := runner.ESHost
+	parsedES, err := url.Parse(runner.ESHost)
+	require.NoError(runner.T(), err)
+	if parsedES.Port() == "" {
+		fixedESHost = fmt.Sprintf("%s:443", fixedESHost)
+	}
+
+	runner.T().Logf("configuring beats with %s / %s", fixedESHost, fixedKibanaHost)
+
+	testUuid, err := uuid.NewV4()
+	require.NoError(runner.T(), err)
+	runner.testUuid = testUuid.String()
+	parsedCfg := fmt.Sprintf(beatOutConfig, fixedESHost, apiResp.Id, apiResp.APIKey, fixedKibanaHost, testUuid.String())
+	err = runner.agentFixture.WriteFileToWorkDir(ctx, parsedCfg, fmt.Sprintf("%s.yml", runner.testbeatName))
+	require.NoError(runner.T(), err)
+}
+
+// run the beat with default metricsets, ensure no errors in logs + data is ingested
+func (runner *BeatRunner) TestRunAndCheckData() {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute*4)
+	defer cancel()
+
+	// in case there's already a running template, delete it, forcing the beat to re-install
+	_ = estools.DeleteIndexTemplatesDataStreams(ctx, runner.requirementsInfo.ESClient, fmt.Sprintf("*%s*", runner.testbeatName))
+
+	err := runner.agentFixture.RunBeat(ctx)
+	require.NoError(runner.T(), err)
+
+	docs, err := estools.GetLatestDocumentMatchingQuery(ctx, runner.requirementsInfo.ESClient, map[string]interface{}{
+		"match": map[string]interface{}{
+			"host.test-id": runner.testUuid,
+		},
+	}, fmt.Sprintf("*%s*", runner.testbeatName))
```

Comment:
We are only doing it once in the whole test, let's keep it simple and allocate one extra string ;)
```suggestion
	docs, err := estools.GetLatestDocumentMatchingQuery(ctx, runner.requirementsInfo.ESClient, map[string]interface{}{
		"match": map[string]interface{}{
			"host.test-id": runner.testUuid,
		},
	}, "*"+runner.testbeatName+"*")
```